File: dfs.java
import java.util.ArrayList;
import java.util.Scanner;
public class dfs {
    private void DFSrec(int node,ArrayList<ArrayList<Integer>> adj,ArrayList<Integer> ans, ArrayList<Boolean> visited){
        visited.set(node,true);
        ans.add(node);
        for(int j=0;j<adj.get(node).size();j++){
            int neigh=adj.get(node).get(j);
            if(!visited.get(neigh)){
                DFSrec(neigh, adj,ans,visited);
            }
        }

    }
    public ArrayList<Integer> dfsofgraph(int v,ArrayList<ArrayList<Integer>> adj,ArrayList<Integer> ans,ArrayList<Boolean> visited){
        DFSrec(0,adj,ans,visited);
        return ans;
    }
    public static void display(int vertex,ArrayList<ArrayList<Integer>> adj){
        for(int i=0;i<vertex;i++){
            System.out.println(i + "->" + " ");
            for(int j=0;j<adj.get(i).size();j++){
                System.out.print(adj.get(i).get(j) + " ");
            }
        }
    }
    public static void main(String[] args) {
        dfs SS = new dfs();
        Scanner sc = new Scanner(System.in);
        int vertex,edges;
        System.out.println("Enter Vertex Length: ");
        vertex=sc.nextInt();
        System.out.println("Enter Edge Length: ");
        edges=sc.nextInt();
        ArrayList<ArrayList<Integer>> AdjList= new ArrayList<>();
        ArrayList<Integer> ans = new ArrayList<>();
        ArrayList<Boolean> visited = new ArrayList<>();
        for (int i=0; i<vertex; i++){
            AdjList.add(new ArrayList<>());
            visited.add(false);
        }
        int u,v;
        for(int i=0;i<edges;i++){
            u=sc.nextInt();
            v=sc.nextInt();
            AdjList.get(u).add(v);
            AdjList.get(v).add(u);
        }
        display(vertex,AdjList);
        SS.dfsofgraph(vertex,AdjList,ans,visited);
        display(vertex, AdjList);
        System.out.println("DFS Traversal: " + ans);
        sc.close();
    }       
}

File: dfs.cpp
#include<iostream>
#include<vector>
using namespace std;
class dfs{
    public:
    void DFSrec(int node,vector<int>adj[],vector<int>ans,vector<bool>visited){
        visited[0]=1;
        ans.push_back(node);
        for(int j=0;j<adj[node].size();j++){
            if(!visited[adj[node][j]]){
                DFSrec(adj[node][j],adj,ans,visited);
            }
        }
    }
    vector<int> dfsofgraph(int v,vector<int>adj[]){
        vector<int>ans;
        vector<bool>visited(v,0);
        DFSrec(0,adj,ans,visited);
        return ans;
    }
    void display(int vertex,vector<int>AdjList[]){
        for(int i=0;i<vertex;i++){
            cout<<i<<"->";
            for(int j=0;j<AdjList[i].size();j++){
                cout<<AdjList[i][j]<<" ";
            }
            cout<<endl;
        } 
    }
};
int main(){
    dfs SS;
    int vertex,edges;
    cout<<"Enter Vertex length: ";
    cin>>vertex;
    cout<<"Enter Edge length: ";
    cin>>edges;
    vector<int>AdjList[vertex];
    int u,v;
    for(int i=0;i<edges;i++){
        cin>>u>>v;
        AdjList[u].push_back(v);
        AdjList[v].push_back(u);
    }
    SS.display(vertex,AdjList);
    SS.dfsofgraph(vertex,AdjList);
    SS.display(vertex,AdjList);
}